File: 2429-minimize-xor/2429-minimize-xor.cpp
class Solution {
    // 0011001
    // 1001000
public:
    int minimizeXor(int num1, int num2) {
        auto x = num1;
        auto toAddInX = __builtin_popcount(num2) - __builtin_popcount(x);
        cout<<toAddInX<<endl;
        
        if (toAddInX > 0) {
            int traverse = 0;
            while (toAddInX) {
                if (x & (1 << traverse)) {} else {
                    x |= 1 << traverse;
                    toAddInX--;
                }
                traverse++;
            }
        } 
        
        if (toAddInX < 0) {
            int traverse = 0;
            while (toAddInX != 0) {
                if (x & (1 << traverse)) {
                    x ^= 1 << traverse;
                    toAddInX++;
                }
                traverse++;
            }
        } 
        
        return x;
    }
};

File: leetcode/2429-minimize-xor/2429-minimize-xor.rs
impl Solution {
    pub fn minimize_xor(num1: i32, num2: i32) -> i32 {
        let (mut traverse, mut x) = (0, num1);
        let mut to_add_in_x = num2.count_ones() as i32 - num1.count_ones() as i32;
        while to_add_in_x != 0 {
            let is_one = x & (1 << traverse);
            if to_add_in_x > 0 && is_one == 0 {
                x |= 1 << traverse;
                to_add_in_x -= 1;
            }
            if to_add_in_x < 0 && is_one != 0 {
                x ^= 1 << traverse;
                to_add_in_x += 1;
            }
            traverse += 1;
        }
        x 
    }
}